### PR TITLE
Habitat test expansion

### PR DESCRIPTION
**Pull request by tpowell-progress**
## Description
In order to support a Habitat-only pipeline, tests need to be expanded to run against habitat.

Windows tests will be broken until the next push of a hab package so there are currently prevented from triggering errors.

## Related Issue
CHEF-15034

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by johnmccrae — APPROVED** [review]

File: habitat/tests/test.sh
Comment (johnmccrae) [inline habitat/tests/test.sh:1]: Here you have test.sh as the file name but for windows it's spec.ps1. Should the file names be similar? spec.sh and spec.ps1?

File: spec/integration/client/open_ssl_spec.rb
Comment (johnmccrae) [inline spec/integration/client/open_ssl_spec.rb:1]: Should the windows openssl version still be at 1.0.2zi? Isn't main supposed to be 3.x.x?

File: habitat/tests/test.sh
Comment (tpowell-progress) [reply habitat/tests/test.sh:1]: For some reason, the Windows pipeline has a `habitat-test.ps1` which calls `spec.ps1` but Linux contains it all in `test.sh`

File: spec/integration/client/open_ssl_spec.rb
Comment (tpowell-progress) [reply spec/integration/client/open_ssl_spec.rb:1]: last i left this PR it was 3.3.x but now it should be 3.0.9. That said, this was previously _also_ disabled.

**Review by johnmccrae — APPROVED** [review]

**Pull request merged**